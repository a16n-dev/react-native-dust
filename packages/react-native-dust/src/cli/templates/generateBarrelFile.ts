import { GeneratedProject } from './generatedProject.js';
import type { codegenOptions } from '../core/codegenTypes.js';

/**
 * Adds a barrel file to the given generated project. This should reexport all the
 * required modules for the package.
 */
export function generateBarrelFile(
  project: GeneratedProject,
  opts: codegenOptions
) {
  const file = project.addSourceFile('index.ts');

  // Add exports
  file.addExportDeclaration({ moduleSpecifier: './theme' });
  file.addExportDeclaration({ moduleSpecifier: './tokens' });
  if (opts.config.options.generatePlatformHelpers) {
    file.addExportDeclaration({ moduleSpecifier: './utilities' });
  }
}
