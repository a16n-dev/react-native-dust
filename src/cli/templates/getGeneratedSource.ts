import { Project, SourceFile, ts } from 'ts-morph';
import ModuleKind = ts.ModuleKind;
import ScriptTarget = ts.ScriptTarget;

export interface generatedFile {
  name: string;
  content: string;
}

/**
 * Creates a new in-memory TS file with some preconfigured compiler options.
 */
export function createSourceFile(fileName: string): SourceFile {
  const project = new Project({
    useInMemoryFileSystem: true,
    compilerOptions: {
      declaration: true,
      module: ModuleKind.ESNext,
      target: ScriptTarget.ES2020,
    },
  });
  return project.createSourceFile(fileName, '', { overwrite: true });
}

/**
 * Takes a generated TS file, and returns the emitted .js & .d.ts files
 */
export function getGeneratedSource(file: SourceFile): generatedFile[] {
  const output = file.getEmitOutput({
    emitOnlyDtsFiles: false,
  });

  return output.getOutputFiles().map((file) => ({
    name: file.getFilePath(),
    content: file.getText(),
  }));
}
