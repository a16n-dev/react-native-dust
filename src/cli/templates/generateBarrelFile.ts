import { GeneratedProject } from './getGeneratedSource';

/**
 * Adds a barrel file to the given generated project. This should reexport all the
 * required modules for the package.
 */
export function generateBarrelFile(project: GeneratedProject) {
  const file = project.addSourceFile('index.ts');

  // Add exports
  file.addExportDeclaration({ moduleSpecifier: './theme' });
  file.addExportDeclaration({ moduleSpecifier: './tokens' });
  file.addExportDeclaration({ moduleSpecifier: './utilities' });
}
