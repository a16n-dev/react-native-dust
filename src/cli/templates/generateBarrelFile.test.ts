import { describe, expect, test } from 'vitest';
import { generateBarrelFile } from './generateBarrelFile';
import { GeneratedProject } from './getGeneratedSource';

describe('generateBarrelFile', () => {
  test('output does not change unexpectedly', () => {
    const project = new GeneratedProject();

    generateBarrelFile(project);

    expect(project.getGeneratedFiles()).toMatchSnapshot();
  });
});
