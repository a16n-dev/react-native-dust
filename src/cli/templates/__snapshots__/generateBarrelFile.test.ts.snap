// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateBarrelFile > output does not change unexpectedly 1`] = `
[
  {
    "content": "export * from "./theme";
export * from "./tokens";
export * from "./utilities";
",
    "name": "./index.js",
  },
  {
    "content": "export * from "./theme";
export * from "./tokens";
export * from "./utilities";
",
    "name": "./index.d.ts",
  },
]
`;
